VictoriaAlbert/VictoriaAlbert/APIRequestManager.swift

```diff
@@ -0,0 +1,63 @@
+//
+//  APIRequestManager.swift
+//  VictoriaAlbert
+//
+//  Created by Tong Lin on 11/10/16.
+//  Copyright © 2016 Tong Lin. All rights reserved.
+//
+
+import Foundation
+
+internal class APIRequestManager {
+    internal static let manager: APIRequestManager = APIRequestManager()
+    init() {}
+    
+    func getObjects(keyWords: String, callback: @escaping ((Data?)->Void)){
+        let url: URL = URL(string: "http://www.vam.ac.uk/api/json/museumobject/search?q=\(keyWords)")!
+        
+        let session: URLSession = URLSession(configuration: URLSessionConfiguration.default)
+        
+        session.dataTask(with: url){ (data: Data?, response: URLResponse?, error: Error?) in
+            if error != nil{
+                print(error!)
+            }
+            
+            if data != nil{
+                print("We have valid data here")
+                callback(data)
+            }
+        }.resume()
+    }
+    
+    func downloadImage(imageID: String, switchPic: Int, callback: @escaping ((Data?)->Void)){
```

Comment:
Since you went down the path of embedding urls in here, I still like the way you thought about factoring this image call into one method.